test/AffInterface.purs

```diff
@@ -97,14 +111,105 @@ testOgmiosDatumCacheFetcher =
 
 testGetChainTip :: Aff Unit
 testGetChainTip = do
-  flip runQueryM (getChainTip *> pure unit) =<< traceQueryConfig
+  flip runQueryM (void getChainTip) =<< traceQueryConfig
 
 testFromOgmiosAddress :: OgmiosAddress -> Aff Unit
 testFromOgmiosAddress testAddr = do
-  liftEffect $ case ogmiosAddressToAddress testAddr of
+  liftEffect case ogmiosAddressToAddress testAddr of
     Nothing -> throw "Failed Address loop"
     Just addr -> addressToOgmiosAddress addr `shouldEqual` testAddr
+
+testGetEraSummaries :: Aff Unit
+testGetEraSummaries = do
+  flip runQueryM (void getEraSummaries) =<< traceQueryConfig
+
+testGetCurrentEpoch :: Aff Unit
+testGetCurrentEpoch = do
+  flip runQueryM (void getCurrentEpoch) =<< traceQueryConfig
+
+testGetSystemStart :: Aff Unit
+testGetSystemStart = do
+  flip runQueryM (void getSystemStart) =<< traceQueryConfig
+
+testPosixTimeToSlot :: Aff Unit
+testPosixTimeToSlot = do
+  traceQueryConfig >>= flip runQueryM do
+    eraSummaries <- getEraSummaries
+    sysStart <- getSystemStart
+    let
+      -- Tests currently pass "exactly" for seconds precision, which makes sense
+      -- given converting to a Slot will round down to the near slot length
+      -- (mostly 1s). If it rounds down and is the end slot, then a check is in
+      -- place that any "extra" time is zero.
+      -- We can allow for Millseconds as (off chain) input if we assume
+      -- the seconds provided by Ogmios are exact, which seems to be the case
+      -- here https://cardano.stackexchange.com/questions/7034/how-to-convert-posixtime-to-slot-number-on-cardano-testnet/7035#7035
+      -- `timeWhenSlotChangedTo1Sec = POSIXTime 1595967616000` - exactly
+      -- divisible by 1 second.
+      posixTimes = mkPosixTime <$>
+        [ "1603636353000"
+        , "1613636755000"
+        , "1753645721000"
+        ]
+    traverse_ (idTest eraSummaries sysStart identity) posixTimes
+    -- With Milliseconds, we generally round down, provided the aren't at the
+    -- end  with non-zero excess:
+    idTest eraSummaries sysStart
+      (const $ mkPosixTime "1613636754000")
+      (mkPosixTime "1613636754999")
+    idTest eraSummaries sysStart
+      (const $ mkPosixTime "1613636754000")
+      (mkPosixTime "1613636754500")
+    idTest eraSummaries sysStart
+      (const $ mkPosixTime "1613636754000")
+      (mkPosixTime "1613636754499")
+  where
+  idTest
+    :: EraSummariesQR
+    -> SystemStartQR
+    -> (POSIXTime -> POSIXTime)
+    -> POSIXTime
+    -> QueryM Unit
+  idTest es ss transf posixTime = liftEffect do
+    posixTimeToSlot es ss posixTime >>= case _ of
+      Left err -> throw $ show err
+      Right slot -> do
+        ePosixTime <- slotToPosixTime es ss slot
+        either (throw <<< show) (shouldEqual $ transf posixTime) ePosixTime
+
+  mkPosixTime :: String -> POSIXTime
+  mkPosixTime = POSIXTime <<< unsafePartial fromJust <<< BigInt.fromString
+
+testSlotToPosixTime :: Aff Unit
+testSlotToPosixTime = do
```

Comment:
How do you feel about introducing Quickcheck for tests? Probably out of scope for this PR, but generally speaking it would be great to start building out a good support for quickcheck in CTL.

Reply:
I made an issue for that
https://github.com/Plutonomicon/cardano-transaction-lib/issues/466

Reply:
Great! Resolving this and tackling in #466